clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -672,7 +672,11 @@ public void onHeartbeatRequestSent() {
         MemberState state = state();
         if (isStaled()) {
             log.debug("Member {} is staled and is therefore leaving the group.  It will rejoin upon the next poll.", memberEpoch);
-            transitionToJoining();
+            // clear the current assignment and subscription, and trigger rebalance listener on the next poll
+            invokeOnPartitionsRevokedCallback(subscriptions.assignedPartitions()).whenComplete((r, e) -> {
```

Comment:
when the member becomes staled, it is still part of the group because it hasn't left yet.  however, the background thread will need to leave the group and revoke all of its assignment because we assume the consumer is not being used - this is the logic for max.poll.interval.ms.  since we are leaving the group, we will need to revoke all assignments.

Reply:
Isn't revocation something that the broker tells the member to because of rebalancing? In this case it is the member that decides to leave the group because it exceeded the poll interval. It is similar to close. I do not think we revoke partitions in close, do we?

Reply:
OK, I just saw https://issues.apache.org/jira/browse/KAFKA-15696

Reply:
Revocation at `close()`-time seems legit.

Reply:
I discussed this offline with @dajac and we think that since exceeding the poll interval is something unexpected we should rather call `onPartitionLost()` rather than `onPartitionRevoked()`.

Reply:
I agree.  We had a bit of discussion around this.  I think whether calling onPartitionLost/Revoke is up for discussion. Perhaps we limit the scope of the change to just clearing the assignment subscription w/o invoking the callbacks.

Reply:
Yes, good idea! I agree!